src/Client.re

```diff
@@ -26,7 +34,9 @@ let start =
       | Incoming.Ready =>
         Log.info("Ready");
 
-        send(Outgoing.Initialize({requestId: 1, initData}));
+        incr(lastRequestId);
+        let requestId = lastRequestId^;
+        send(Outgoing.Initialize({requestId, initData}));
```

Comment:
This is unnecessary indirection, and there doesn't seem to be any benefit as record fields can be assigned to directly. It's just polluting the namespace and adding more things to keep track of. Which probably doesn't seem so bad in such a small scope, but code evolves and grows, and as it grows the problem grows with it. The number of possible states grows exponentially with the number of variables.